### PR TITLE
oat++ suite bump to version 1.1

### PR DESCRIPTION
**Pull request by mheyman**
**Oat++ suite bump to version 1.1**

Oat++ bumped to version 1.1 in mid May. We've been using the core, websockets, and swagger for a month on both Windows and Linux with no issues.

### CODE REVIEW

File: ports/oatpp-consul/CONTROL [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Source: oatpp-consul
-Version: 1.0.0
+Version: 1.1.0
 Description: OAT++ Modern web framework consul module
```
Comment (NancyLi1013) [inline ports/oatpp-consul/CONTROL:3]: Could you please add homepage here?

File: ports/oatpp-curl/CONTROL [hunk 2]
```diff
@@ -1,4 +1,4 @@
 Source: oatpp-curl
-Version: 1.0.0
+Version: 1.1.0
 Description: Oat++ Modern web framework curl module to use libcurl as a RequestExecutor on the oatpp's ApiClient
```
Comment (NancyLi1013) [inline ports/oatpp-curl/CONTROL:3]: Could you please add homepage here?

File: ports/oatpp-libressl/CONTROL [hunk 3]
```diff
@@ -1,4 +1,4 @@
 Source: oatpp-libressl
-Version: 1.0.0
+Version: 1.1.0
 Build-Depends: libressl, oatpp
 Description: Oat++ libressl module providing secure server and client connection providers
```
Comment (NancyLi1013) [inline ports/oatpp-libressl/CONTROL:4]: The same as above.

File: ports/oatpp-mbedtls/CONTROL [hunk 4]
```diff
@@ -1,4 +1,4 @@
 Source: oatpp-mbedtls
-Version: 1.0.0
+Version: 1.1.0
 Build-Depends: mbedtls, oatpp
 Description: Oat++ Mbed TLS submodule providing secure server and client connection providers
```
Comment (NancyLi1013) [inline ports/oatpp-mbedtls/CONTROL:4]: The same as above.

File: ports/oatpp-swagger/CONTROL [hunk 5]
```diff
@@ -1,4 +1,4 @@
 Source: oatpp-swagger
-Version: 1.0.0
+Version: 1.1.0
 Description: Oat++ OpenApi (Swagger) UI submodule
```
Comment (NancyLi1013) [inline ports/oatpp-swagger/CONTROL:3]: The same as above.

File: ports/oatpp-websocket/CONTROL [hunk 6]
```diff
@@ -1,4 +1,4 @@
 Source: oatpp-websocket
-Version: 1.0.0
+Version: 1.1.0
 Description: Oat++ websocket module
```
Comment (NancyLi1013) [inline ports/oatpp-websocket/CONTROL:3]: The same as above.

File: ports/oatpp/CONTROL [hunk 7]
```diff
@@ -1,3 +1,3 @@
 Source: oatpp
-Version: 1.0.0
+Version: 1.1.0
 Description: Modern web framework
```
Comment (NancyLi1013) [inline ports/oatpp/CONTROL:3]: The same as above.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
The failures on x86-windows, x64-windows, x64-windows-static are like this:
```
-- Configuring done
CMake Error at C:/agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:326 (_add_library):
  Target "oatpp-mbedtls" links to target "mbedtls::TLS" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  src/CMakeLists.txt:2 (add_library)


CMake Error at C:/agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:326 (_add_library):
  Target "oatpp-mbedtls" links to target "mbedtls::X509" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  src/CMakeLists.txt:2 (add_library)


CMake Error at C:/agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:326 (_add_library):
  Target "oatpp-mbedtls" links to target "mbedtls::Crypto" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  src/CMakeLists.txt:2 (add_library)
```

File: ports/oatpp-consul/CONTROL (hunk 1 above)
Comment (mheyman) [reply ports/oatpp-consul/CONTROL:3]: Done (didn't know about "Homepage:")

File: ports/oatpp-curl/CONTROL (hunk 2 above)
Comment (mheyman) [reply ports/oatpp-curl/CONTROL:3]: Done (didn't know about "Homepage:")

File: ports/oatpp-libressl/CONTROL (hunk 3 above)
Comment (mheyman) [reply ports/oatpp-libressl/CONTROL:4]: Done (didn't know about "Homepage:")

File: ports/oatpp-mbedtls/CONTROL (hunk 4 above)
Comment (mheyman) [reply ports/oatpp-mbedtls/CONTROL:4]: Done (didn't know about "Homepage:")

File: ports/oatpp-swagger/CONTROL (hunk 5 above)
Comment (mheyman) [reply ports/oatpp-swagger/CONTROL:3]: Done (didn't know about "Homepage:")

File: ports/oatpp-websocket/CONTROL (hunk 6 above)
Comment (mheyman) [reply ports/oatpp-websocket/CONTROL:3]: Done (didn't know about "Homepage:")

File: ports/oatpp/CONTROL (hunk 7 above)
Comment (mheyman) [reply ports/oatpp/CONTROL:3]: Done (didn't know about "Homepage:")

**Comment by NancyLi1013** [comment]
Please also look into the regressions and try to fix them.

**Comment by mheyman** [comment]
The latest code has a change in case for a find_package() that was not failing on my build but seemed to be the issue on yours. We'll see ;-)

**Comment by NancyLi1013** [comment]
I will try to reproduce this on my local.

**Comment by NancyLi1013** [comment]
It seems to be related with case sensitive.

I noticed that it used `if(MBEDTLS_FOUND)` in 
`cmake/module/Findmbedtls.cmake
`.

But from the configure log, I found the info is:
`-- Found mbedTLS: D:/installed/x64-windows/include (found version "2.16.3")`

So could you please try to update `MBEDTLS_FOUND` as `mbedTLS_FOUND` and rebuild this on CI?

**Comment by BillyONeal** [comment]
PR validation still looks like legit failures

**Comment by NancyLi1013** [comment]
@mheyman 
Is work still being done for this PR?

**Comment by mheyman** [comment]
I still want to be able to use oatpp from github vcpkg. Been too busy to do
more. I have to tell you that we've been using it from our own fork of
vcpkg for a while on both windows and linux and haven't been able to
reproduce the issues it has been having building on your servers and it is
hard to debug (or at least requires arcane knowledge I don't posses). Is
there a faq or other document somewhere listing common steps to take when
your code builds without a problem locally but fails in the azure system
you are using?

Thanks

**Comment by mheyman** [comment]
I think I finally figured out what was going on - there is a version of FindMbedtls.cmake somewhere on the Azure build servers that doesn't exist in our systems. Cmake calls that find code instead of the one that comes with Oat++. And, that find code doesn't set up the mbedtls targets expected by Oat++, just some variables (which I used instead). FYI: there are also log lines showing that pthreads4w is getting pulled in on the Azure build servers and that doesn't get pulled in on our servers either but that may be part of the FindMbedtls.cmake hijacking going on.

**Comment by BillyONeal** [comment]
> there is a version of FindMbedtls.cmake somewhere on the Azure build servers

https://github.com/microsoft/vcpkg/blob/master/ports/mbedtls/vcpkg-cmake-wrapper.cmake is that override.

**Comment by mheyman** [comment]
> > there is a version of FindMbedtls.cmake somewhere on the Azure build servers
> 
> https://github.com/microsoft/vcpkg/blob/master/ports/mbedtls/vcpkg-cmake-wrapper.cmake is that override.

:-) Got it to work using that one or the one that comes with Oat++ for the rest of us

**Comment by KaungZawHtet** [comment]
1.2.0 is coming

**Comment by BillyONeal** [comment]
@NancyLi1013 is on vacation and I think she was waiting for a clean CI run anyway; it's passing now so I merged it.

Thanks for your contribution!

**Comment by NancyLi1013** [comment]
Thanks for your help and follow up @BillyONeal .

**Comment by KaungZawHtet** [comment]
it is still 1.0.0 when I do `vcpkg search oatpp`

95308636-2d2b2280-08b0-11eb-81a3-ba82b82e34b4

**Comment by BillyONeal** [comment]
> it is still 1.0.0 when I do `vcpkg search oatpp`
> 
> 95308636-2d2b2280-08b0-11eb-81a3-ba82b82e34b4

Did you do a `git pull` to update your vcpkg repo?

**Comment by KaungZawHtet** [comment]
opps! I notice what I missed. Thx

**Pull request merged**